pkgs/development/interpreters/php/copy-libtool-files-from-packages.diff

```diff
@@ -0,0 +1,45 @@
+diff --git a/scripts/phpize.in b/scripts/phpize.in
+index 2f117656..6e5cd718 100644
+--- a/scripts/phpize.in
++++ b/scripts/phpize.in
+@@ -9,9 +9,13 @@ includedir="`eval echo @includedir@`/php"
+ builddir="`pwd`"
+ SED="@SED@"
+ 
+-FILES_BUILD="php.m4 shtool libtool.m4 ax_check_compile_flag.m4 ax_gcc_func_attribute.m4 php_cxx_compile_stdcxx.m4 pkg.m4 \
+-    config.guess config.sub ltmain.sh Makefile.global gen_stub.php"
++FILES_BUILD="php.m4 shtool ax_check_compile_flag.m4 ax_gcc_func_attribute.m4 php_cxx_compile_stdcxx.m4 \
++    Makefile.global gen_stub.php"
++auxdir="@libtool@/share/libtool/build-aux"
++AUX_FILES="config.sub config.guess ltmain.sh"
+ FILES="run-tests*.php"
++aclocaldir="@libtool@/share/aclocal"
++LIBTOOL_FILES="libtool.m4 ltoptions.m4 ltsugar.m4 ltversion.m4 lt~obsolete.m4"
+ CLEAN_FILES="$FILES *.o *.lo *.la .libs/ build/ modules/ \
+   config.nice configure configure.ac \
+   config.h config.h.in conftest* libtool config.cache autom4te.cache/ \
+@@ -149,6 +153,9 @@ phpize_copy_files()
+   test -d build || mkdir build
+ 
+   (cd "$phpdir" && cp $FILES_BUILD "$builddir"/build)
++  (cd "$aclocaldir" && cp $LIBTOOL_FILES "$builddir"/build)
++  (cp "@pkg-config@/share/aclocal/pkg.m4" "$builddir"/build)
++  (cd "$auxdir" && cp $AUX_FILES "$builddir"/build)
+   (cd "$phpdir" && cp $FILES "$builddir")
+ }
+ 
+diff --git a/scripts/phpize.m4 b/scripts/phpize.m4
+index be7065b7..6fb4f337 100644
+--- a/scripts/phpize.m4
++++ b/scripts/phpize.m4
+@@ -5,6 +5,10 @@ dnl comments starting with # and empty newlines from the included files.
+ m4_include([build/ax_check_compile_flag.m4])
+ m4_include([build/ax_gcc_func_attribute.m4])
+ m4_include([build/libtool.m4])
++m4_include([build/ltoptions.m4])
++m4_include([build/ltsugar.m4])
++m4_include([build/ltversion.m4])
++m4_include([build/lt~obsolete.m4])
```

Comment:
How will these work if you remove them?

Reply:
Never mind, phpize script will use this to create configure.ac:

https://github.com/php/php-src/blob/64e2832bc8065f24c4d273075608f33826c71fd8/scripts/phpize.in#L159


and it will also copy the necessary files into the extension's build directory:

https://github.com/php/php-src/blob/64e2832bc8065f24c4d273075608f33826c71fd8/scripts/phpize.in#L151

would be nice to mention this in the commit message since it is non-trivial.